src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -30,6 +30,8 @@ public struct MetricPoint
 
         private HistogramBuckets histogramBuckets;
 
+        private Base2ExponentialBucketHistogram exponentialBucketHistogram;
```

Comment:
won't this increase the size of the MetricPoint struct, and hence affect overall memory, as we pre-allocate 2k of MP?

I am facing similar challenge for Exemplars too (as storing pointer to Exemplar increases MP size!), and as a solution, I am trying to bundle histogramBuckets and exemplars into a new class, and keep its instance in MP, so MP's size remains same.

Reply:
I could try combining HistogramBuckets and Base2ExponentialBucketHistogram into a single class to keep size of MP the same.

Though, do you think it makes sense to further combine exemplars into the same class as histogram buckets? Coupling the memory consumption of enabling exemplars with the recording of a histogram seems a little weird.

Reply:
> I could try combining HistogramBuckets and Base2ExponentialBucketHistogram into a single class to keep size of MP the same.

^ would be good.

Reply:
> , do you think it makes sense to further combine exemplars into the same class as histogram buckets?

Most likely yes, but I'll give better answer once I finish my prototype (expect PRs on it in next day or so).